app/controllers/v2/policies_controller.rb

```diff
@@ -54,6 +54,21 @@ def destroy
     permission_for_action :destroy, Rbac::POLICY_DELETE
     permitted_params_for_action :destroy, { id: ID_TYPE }
 
+    def systems
+      hosts_added, hosts_removed = compliance_policy.bulk_assign(
+        V2::PolicyHost, permitted_params[:data]&.dig(:hosts)
```

Comment:
Let's simplify things for among others better type inference:
```suggestion
        V2::PolicyHost, V2::Host.find(permitted_params[:data]&.dig(:hosts))
```
This should fail if at least one of the IDs is not found and then you don't need the reflection below.

Reply:
This also allows to wrap things around a pundit policy 😉 so we can sort out inventory groups easily.

Reply:
The only caveat this causes is that using an empty array of host ids will not pass this `find` method. 

I could make so that it does, but that would make the `systems` controller method longer which kinda goes against our "fat model, skinny controller" goal.

I could also move this find method to the model level and find the model in which we `find` from reflections.

Reply:
Hmm, I guess we can handle that separately? Let me think about it